### DISCUSSION TITLE
Setting default height for ViewContainerPart

### DISCUSSION THREAD

**Discussion by hx8563** [eclipse-theia/theia#7728] [General] [answered]
### Feature Description:
Is there a way to specify an initial height for every ViewContainerPart in ViewContainer instead of having all parts with same height?

**Comment by vince-fugnitto** [comment, marked as answer]
@hx8563 thank you for the issue, please post questions on [discussions](https://github.com/eclipse-theia/theia/discussions) in the future.
Are you looking to create a new `ViewContainerPart` or are you attempting to update existing containers?

**Reply by hx8563** [reply]
Thanks for reply~
I have a custom page in my extension, which has several child page organized in a ViewContainer. 
I want to find if there is a way to set a default height scale for each child page, like what  'flex-grow' dose.

**Reply by kittaakos** [reply]
Does this help?

https://github.com/eclipse-theia/theia/blob/0a691e224cec050a7f51f99d12924ebf478fc4c7/packages/core/src/browser/view-container.ts#L630

We use it for the Debug view, for instance:

https://github.com/eclipse-theia/theia/blob/0a691e224cec050a7f51f99d12924ebf478fc4c7/packages/debug/src/browser/view/debug-session-widget.ts#L91-L95

**Reply by hx8563** [reply]
I will try, thanks!

**Reply by hx8563** [reply]
<img width="386" alt="屏幕快照 2020-05-06 下午3 54 07" src="https://user-images.githubusercontent.com/6118926/81150067-e6a86380-8fb1-11ea-94ae-aa5279a2c7d0.png">

@kittaakos I have tried to set the weight in addWidget, but it seems like each sub-view in ViewContainer is remain with the same height, and the sub-view in debugger is also displayed with equal height(like the screenshot above), do I need some addition configuration?

**Reply by kittaakos** [reply]
No, it seems to be a bug in the view containers. Can you please file a bug report? Then the issue will get slightly better visibility in the community, and hopefully someone will fix it. Thank you!

**Reply by hx8563** [reply]
Thanks for your help, I have opened a issue: #7755
